parallel-consumer-examples/parallel-consumer-example-core/src/test/java/io/confluent/parallelconsumer/examples/core/CoreAppMetricsIntegrationTest.java

```diff
@@ -0,0 +1,100 @@
+package io.confluent.parallelconsumer.examples.core;
+
+/*-
+ * Copyright (C) 2020-2022 Confluent, Inc.
+ */
+
+import io.confluent.csid.utils.LongPollingMockConsumer;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.clients.producer.MockProducer;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.Serdes;
+import org.awaitility.Awaitility;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import pl.tlinkowski.unij.api.UniLists;
+
+import java.io.BufferedInputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.when;
+import static pl.tlinkowski.unij.api.UniLists.of;
+
+@Slf4j
+@Testcontainers
+public class CoreAppMetricsIntegrationTest{
```

Comment:
does this work without needing the docker compose etc?

Reply:
no docker-compose needed, just TestContainers